models/Asset.php

```diff
@@ -873,6 +873,14 @@ public function getFrontendFullPath()
         $path = urlencode_ignore_slash($path);
 
         $prefix = \Pimcore::getContainer()->getParameter('pimcore.config')['assets']['frontend_prefixes']['source'];
+        if (!$prefix) {
+            $prefix = sprintf(
```

Comment:
This would add the scheme and hostname even if not necessary, e.g. in frontend templates. On big pages this could result in a significant bigger response body. Maybe it's better to just hande this in the `AssetController`?

Reply:
yes, I also think it should be handled there, however, in the `AssetController` I can only validate it there is a schema, or check it this `['frontend_prefixes']['source']` is not empty

Reply:
I'd check for a schema with a regex, because the frontend path could be also changed by an event and not only by config 😊

Reply:
ok, going to close this one and open a new MR, thanks